test/secret-storage/SecretStore.ts

```diff
@@ -0,0 +1,1261 @@
+        const timestamp = await time.latest() * 1000;
+        let signTimestamp = await time.latest() - 700;
+        let [attestationSign, attestation] = await createAttestation(
+            pubkeys[15],
+            image2,
+            wallets[14],
+            timestamp - 540000
+        );
+
+        let storageCapacity = 1e9;
+        let signedDigest = await createSecretStoreSignature(addrs[1], storageCapacity, signTimestamp,
+            wallets[15]);
+
+        await expect(secretStore.connect(signers[1]).registerSecretStore(
+            attestationSign,
+            attestation,
+            storageCapacity,
+            signTimestamp,
+            signedDigest,
+            0
+        )).to.revertedWithCustomError(secretStore, "SecretStoreSignatureTooOld");
+    });
+
+    it("cannot register secret store with different attestation pubkey and digest signing key", async function () {
+        const timestamp = await time.latest() * 1000;
+        let signTimestamp = await time.latest() - 540;
+        let [attestationSign, attestation] = await createAttestation(
+            pubkeys[15],
+            image2,
+            wallets[14],
+            timestamp - 540000
+        );
+
+        let storageCapacity = 1e9;
+        let signedDigest = await createSecretStoreSignature(addrs[1], storageCapacity, signTimestamp,
+            wallets[16]);
+
+        await expect(secretStore.connect(signers[1]).registerSecretStore(
+            attestationSign,
+            attestation,
+            storageCapacity,
+            signTimestamp,
+            signedDigest,
+            0
+        )).to.revertedWithCustomError(secretStore, "SecretStoreInvalidSigner");
+    });
+
+    it("cannot register secret store with same enclave key twice", async function () {
+        const timestamp = await time.latest() * 1000;
+        let signTimestamp = await time.latest() - 540;
+        let [attestationSign, attestation] = await createAttestation(
+            pubkeys[15],
+            image2,
+            wallets[14],
+            timestamp - 540000
+        );
+
+        let storageCapacity = 1e9;
+        let signedDigest = await createSecretStoreSignature(addrs[1], storageCapacity, signTimestamp,
+            wallets[15]);
+        await secretStore.connect(signers[1]).registerSecretStore(
+            attestationSign,
+            attestation,
+            storageCapacity,
+            signTimestamp,
+            signedDigest,
+            0
+        )
+        await expect(secretStore.connect(signers[1]).registerSecretStore(
+            attestationSign,
+            attestation,
+            storageCapacity,
+            signTimestamp,
+            signedDigest,
+            0
+        )).to.revertedWithCustomError(secretStore, "SecretStoreEnclaveAlreadyExists");
+    });
+
+    // drain then deregister with no occupied storage
+    it('can deregister secret store without occupied storage', async function () {
```

Comment:
you can separate out these deregister tests to different describe section, and register one secret store in test setup.

Reply:
For other contracts UTs, we have register and deregister within a single describe block, so did the same here. Updated the block heading.